builtin/credential/cert/path_login.go

```diff
@@ -77,16 +78,21 @@ func (b *backend) pathLogin(
 		ttl = b.System().DefaultLeaseTTL()
 	}
 
+	maxTTL := matched.Entry.MaxTTL
+	if maxTTL == 0 {
+		maxTTL = b.System().MaxLeaseTTL()
```

Comment:
You don't need to do this, the max TTL at the mount/system level will be applied automatically. However, what you *do* need to do is check if maxTTL is zero, and if it isn't, ensure that the TTL value is capped to it.

Basically: if both are zero, leave it at zero. If both exist, ensure that the TTL is less than the MaxTTL. If only TTL exists, leave things alone, and if only max TTL exists, set TTL to the system default and then cap it to max TTL if max TTL is less.